fdbrpc/SimExternalConnection.actor.cpp

```diff
@@ -108,6 +108,57 @@ void MockDNS::clearMockTCPEndpoints() {
 	hostnameToAddresses.clear();
 }
 
+std::string MockDNS::toString() {
+	std::string ret;
+	int i = 0;
+	for (auto it = hostnameToAddresses.begin(); it != hostnameToAddresses.end(); ++it, ++i) {
+		ret += it->first + ',';
+		std::vector<NetworkAddress> addresses = it->second;
+		for (int j = 0; j < addresses.size(); ++j) {
+			ret += addresses[j].toString();
+			if (j != addresses.size() - 1) {
+				ret += ',';
+			}
+		}
+		if (i != hostnameToAddresses.size() - 1) {
+			ret += ';';
+		}
+	}
+	return ret;
+}
+
+MockDNS MockDNS::parseFromString(const std::string& s) {
```

Comment:
Is this function assumes `s` is well formatted and it only contains happy path code? What if `s` is just some random string?

If so, consider adding some format checking and and also some unit test. Can do it in a followup PR.

Reply:
This string can only come from an existing MockDNS being converted to a string first, so this assumption is safe.